### **Data Analytics Task: Customer Feedback Analysis**

**Scenario:**
You work as a data analyst for a company that collects customer feedback across several service categories, including Timeliness, Quality, and Customer Support. Each feedback entry includes a customer ID, the service category, and a rating from 1 to 5 (where 5 is the highest level of satisfaction). Your task is to analyze this feedback to identify areas of strength and opportunities for improvement.

**Objective:**
- Calculate the average rating for each service category.
- Identify the service category with the highest and lowest average rating.
- Determine the distribution of ratings for each category (how many 1s, 2s, etc.).
- Suggest actionable insights based on the analysis.

In [2]:
import pandas as pd

data = pd.read_csv("./modules/customer_feedback.csv")
data

CustomerID          Category  Rating
0       C001        Timeliness       4
1       C002           Quality       5
2       C003  Customer Support       3
3       C004        Timeliness       2
4       C005           Quality       4
5       C006  Customer Support       5
6       C007        Timeliness       5
7       C008           Quality       3
8       C009  Customer Support       2
9       C010        Timeliness       4

In [7]:
category_ratings = data.groupby('Category')['Rating']
category_ratings.mean()

Category
Customer Support    3.333333
Quality             4.000000
Timeliness          3.750000
Name: Rating, dtype: float64

In [13]:
highest_average = category_ratings.mean().idxmax()
lowest_average = category_ratings.mean().idxmin()

print(f"{highest_average}, {lowest_average}")

Quality, Customer Support


In [26]:
# Unstack to fill null values with 0
data.groupby('Category')['Rating'].value_counts().unstack(fill_value=0)

Rating            2  3  4  5
Category                    
Customer Support  1  1  0  1
Quality           0  1  1  1
Timeliness        1  0  2  1

### **Interpretation of Results**

**Average Rating For Each Service Category:**
- Customer Support:   3.33
- Quality:            4.00
- Timeliness:         3.75

Quality has the **highest average** suggesting that the service/product offered by the company is always kept to standard. Customer support has the **lowest average** indicating that after-sales service needs to be worked on, or actionable solutions are hard to relay towards customers. Timeliness is in between both categories, suggesting that the punctuality of employees is at a moderate to decent level.

**Distribution of ratings for each category:**


| Rating	       |1   |2	 |3	  |4   |5   |
|------------------|----|----|----|----|----|
| **Category**     |    |    |    |    |    |	
| Customer Support |0   |1	 |1	  |0   |1   |
| Quality	       |0   |0	 |1	  |1   |1   |
| Timeliness	   |0   |1	 |0	  |2   |1   |

### **Recommendations**
- **Staff Training/Briefing on newer services/products**: A crash course/manual on the newest products/services can increase the quality of After-Sales Care that can be offered to defective items.
- **Quality Control**: As Customer Support is an issue, products that pass through Quality Control can alleviate the need for Customer Support, as maybe upon extended use, the product does not offer the same level of quality.